book/src/Example.md

```diff
@@ -0,0 +1,95 @@
+```
+
+The HostPath driver is configured to create new volumes under `/tmp`
+inside the `csi-pod/hostpath-driver` and thus persist as long as the `csi-pod` itself.
+We can use such volumes in another pod like this:
+```
+kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/docs/master/book/src/example/csi-app.yaml
+```
+
+```
+{{#include example/csi-app.yaml}}
+```
+
+## Troubleshooting
+
+When using `local-cluster-up.sh`, double-check that it gets started with the right features, for example like this:
+```
+RUNTIME_CONFIG=storage.k8s.io/v1alpha1=true \
+ALLOW_PRIVILEGED=1 \
+FEATURE_GATES="BlockVolume=true,MountPropagation=true,CSIPersistentVolume=true" \
+RUNTIME_CONFIG=storage.k8s.io/v1alpha1=true \
+   hack/local-up-cluster.sh
+```
+
+Here is how that cluster looks like after creating the `csi-pod`, the PVC and the example app:
+```
+$ cluster/kubectl.sh get pods
+NAME         READY     STATUS    RESTARTS   AGE
+csi-pod      4/4       Running   0          8m
+my-csi-app   1/1       Running   0          18s
+$ cluster/kubectl.sh get pvc
+NAME      STATUS    VOLUME                                   CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+csi-pvc   Bound     kubernetes-dynamic-pv-aba20e7821f911e8   1Gi        RWO            csi-hostpath-sc   29s
+$ cluster/kubectl.sh get pv
+NAME                                     CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS    CLAIM             STORAGECLASS      REASON    AGE
+kubernetes-dynamic-pv-aba20e7821f911e8   1Gi        RWO            Delete           Bound     default/csi-pvc   csi-hostpath-sc             32s
+$ cluster/kubectl.sh exec -ti my-csi-app /bin/sh
+/ # touch /data/hello-world
+/ # exit
+$ cluster/kubectl.sh exec -ti csi-pod -c hostpath-driver /bin/sh
+/ # find / -name hello-world
+/tmp/aba238a3-21f9-11e8-9164-0242ac110002/hello-world
+```
+
+This error is caused by running on a cluster which lacks the `storage.k8s.io/v1alpha1=true` runtime configuration:
+```
+$ kubectl logs csi-pod external-attacher
+...
+I0306 16:34:50.976069       1 reflector.go:240] Listing and watching *v1alpha1.VolumeAttachment from github.com/kubernetes-csi/external-attacher/vendor/k8s.io/client-go/informers/factory.go:86
+
+E0306 16:34:50.992034       1 reflector.go:205] github.com/kubernetes-csi/external-attacher/vendor/k8s.io/client-go/informers/factory.go:86: Failed to list *v1alpha1.VolumeAttachment: the server could not find the requested resource
+...
+```
+
+The images used by the `csi-pod` are under active development. It can
+happen that they become incompatible with each other. If the
+configuration issues mentioned above have been ruled out, [contact the sig-storage
+team](https://github.com/kubernetes/community/tree/master/sig-storage) and/or
+[run the e2e test](https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#local-clusters):
+```
+go run hack/e2e.go -- --provider=local --test --test_args="--ginkgo.focus=Feature:CSI"
```

Comment:
hack/e2e.sh

Reply:
No, "go run" is really invoked for hack/e2e.go. That's a Go wrapper program which then invokes e2e.sh. See https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#building-kubernetes-and-running-the-tests

Reply:
Got it. Thanks!